common/src/chain/transaction/signature/mod.rs

```diff
@@ -171,29 +155,44 @@ pub fn verify_signature<T: Transactable>(
             }
             Destination::AnyoneCanSpend => {}
         },
-        InputWitness::Standard(witness) => verify_standard_input_signature(
-            chain_config,
-            outpoint_destination,
-            witness,
-            tx,
-            inputs_utxos,
-            input_num,
-        )?,
+        InputWitness::Standard(witness) => {
+            let input_utxo = inputs_utxos.get(input_num).ok_or(
+                DestinationSigError::InvalidInputUtxoIndex(input_num, inputs_utxos.len()),
+            )?;
+
+            let raw_signature = evaluate_witness(input_utxo, witness)?;
```

Comment:
So, this is a major improvement. We're getting there. But I think we can do much better.

I'll come up with a better execution model and come back to you.